app/tests/MinimalApi.Tests/AzureSearchEmbedServiceTest.cs

```diff
@@ -312,4 +315,89 @@ public async Task EmbedImageBlobTestAsync()
             await blobServiceClient.DeleteBlobContainerAsync(blobContainer);
         }
     }
+
+    [Fact]
+    public async Task EnsureTextSplitsOnTinySectionAsync()
```

Comment:
I'd expect a lot more test cases. What about some content with the various word splits or sentence delimiters. Also, consider using a `Theory` to avoid redundant code, and parameterize all the known-inputs, and expected outputs.